### ISSUE TITLE
Story Work

### ISSUE THREAD

**Issue by christiancosgrove** [BoolClub/ProjectCrusade#42]
Now that most of the engine is done, we can (and must) get started on ironing out story specifics. Any ideas, please post them below (including anyone who hasn't yet gotten involved in programming). 

If you're new to the code, I recommend checking out the repository to your computer using [GitHub Desktop](https://desktop.github.com/) and trying out the code by running it in Xamarin. It might give you some ideas of the limitations we're dealing with--again, we're not making an MMORPG here. 

Also, some important things to note about how we've designed in-game events:
- So far, events are triggered by entering an area. These areas can be designated in the rooms we design, which are randomly placed throughout the level.
- In other words, each in-game objective is tied to a room; the room is randomly placed, but it's guaranteed to be in the level (at least in the current system).
- By making the level generation pseudorandom, we add a bit of re-playability. Hopefully gameplay will be fun enough to do this.  
- We will eventually have objectives that can be activated by defeating bosses, for instance, but right now let's focus on what we have with our existing engine.

**Comment by Authman2** [comment]
One idea that I had for the rest rooms was to have a tent inside of them where you could go inside and rest. Also we need to design a battle scene.

**Comment by ConorT0** [comment]
I think that we need to have a concrete idea for our game, and we should iron out all the specifics. Things like how the game should play, how a typical play through should go, and how items should work. We need figure exactly what we want our game to be, then make it to fit that vision.

If the game is going to be as re-playable as possible, we need to have every play through be different. The main source of variety could be items. If we have many different items that all interact in unique ways, it would really add to the replay value. I don't think that we should have a "pick your class" sort of thing, but rather as the player picks up items for a class it gives the player a bonus to items that fall under that class. For example, if the player picks up 3 knight items, they could get a bonus for sword damage or something. However, we should program cool synergies between items from differing classes. If the player from before picks up a wizard staff instead of the 3rd knight item, they would have a wizard staff that shoots knight swords. But if a player with a lot of wizard items picks up a knight sword, they could have an enchanted sword that replaces their staff and can be used to both shoot projectiles and swing in a relatively weak melee attack. 

I think that items should be obtained in two ways. The first way would be the item room that spawns on every floor, and has an item from each class in it. The item room can be opened with some kind of consumable that the player would have to search the floor for. Once in the item room the player can choose one item, and the other disappear. After getting the item, the player must fight the boss to progress to the next floor. 

Once the boss is defeated, a treasure room could spawn for the second source of items. I don't think that the treasure room should spawn every time, and some skill based factor could be used to determine if the room spawns or not. The treasure room contains powerful items that the player trades max health for. This makes players choose between a risky glass cannon build that quickly defeats enemies or a safer high health build.

And of course, all the items would be random. So if on one play-through the player gets super powerful rare item, they will remember it for next time. Also, having items that are randomly generated allows skilled players to beat the game consistently, while a new player can get lucky with a powerful item to carry them to the win.
